src/renderer/utils/utils.ts

```diff
@@ -44,18 +44,50 @@ export const useInterval = (callback: () => void, delay?: number) => {
 };
 
 // Based on https://codeforgeek.com/throttle-function-javascript/
-export const useThrottle = <Type>(
-  fn: (arg: Type) => unknown,
+export const useThrottle = <T extends (...args: any[]) => unknown>(
+  fn: T,
   delay: number
 ) => {
   const lastCalled = useRef<number>(0);
-  return (arg: Type) => {
+  return (...args: Parameters<T>) => {
```

Comment:
Can we wrap the returned function here in a `useCallback`? that way, as long as the provided `fn` and `delay` values don't change, the returned value also won't change.

For context, I noticed while doing work on another feature that the returned value from this function will change every render. Since I was using the returned function as a callback on an event listener, it ended up removing and re-adding the handler on every render (since the function being passed changed).